### PR TITLE
HMS-2694 fix: improve response statuses

### PR DESCRIPTION
**Pull request by frasertweedale**
Improve the HTTP response status for some expected ways host-conf can fail.

- No matching domains (500 -> 404)
- >1 matching domains (500 -> 409)

### CODE REVIEW

**Comment by frasertweedale** [comment]
Your feedback for #18 means we'll touch thiis same code here, so converting to draft and it will probably be superseded later.  Keeping it open for feedback anyway.

File: internal/usecase/repository/host_repository.go [hunk 1]
```diff
@@ -47,13 +49,17 @@ func (r *hostRepository) MatchDomain(db *gorm.DB, options *interactor.HostConfOp
...
 
 	// only one domain is currently supported. Fail if query found multiple doamins.
-	if len(domains) != 1 {
-		return nil, fmt.Errorf("matched %d domains found, only one expected", len(domains))
+	if len(domains) < 1 {
+		msg := fmt.Sprintf("no matching domains")
+		return nil, echo.NewHTTPError(http.StatusNotFound, msg)
+	} else if len(domains) > 1 {
+		msg := fmt.Sprintf("matched %d domains, only one expected", len(domains))
+		return nil, echo.NewHTTPError(http.StatusConflict, msg)
```
Comment (tiran) [inline internal/usecase/repository/host_repository.go:62]: A while ago, I added `NewHTTPErrorF()` to create a HTTP error from a string with formatting.

Comment (frasertweedale) [reply internal/usecase/repository/host_repository.go:62]: I missed that, thanks!

**Comment by frasertweedale** [comment]
Closing.  This change was made part of #18 .

**Pull request closed without merging**